pallets/collator-selection/src/lib.rs

```diff
@@ -664,6 +670,72 @@ pub mod pallet {
         }
     }
 
+    /// Checks if a provided NimbusId SessionKey has an associated AccountId
+    impl<T> AccountLookup<T::AccountId> for Pallet<T>
+    where
+        T: pallet_session::Config + Config,
+        // Implemented only where Session's ValidatorId is directly convertible to collator_selection's ValidatorId
+        <T as Config>::ValidatorId: From<<T as pallet_session::Config>::ValidatorId>,
+    {
+        fn lookup_account(author: &NimbusId) -> Option<T::AccountId>
+        where
+            <T as Config>::ValidatorId: From<<T as pallet_session::Config>::ValidatorId>,
+        {
+            use sp_runtime::traits::Convert;
+            #[allow(clippy::bind_instead_of_map)]
+            pallet_session::Pallet::<T>::key_owner(
+                nimbus_primitives::NIMBUS_KEY_ID,
+                &author.to_raw_vec(),
+            )
+            .and_then(|vid| Some(T::AccountIdOf::convert(vid.into())))
+        }
+    }
+
+    /// Fetch list of all possibly eligible authors to use in nimbus consensus filters
+    ///
+    /// This is currently the static set registered with pallet_session and will be superseded by parachain_staking
+    ///
+    /// NOTE: This should really be in pallet_session as we only use its storage, but since we haven't
+    /// forked that one, this is the next best place.
+    impl<T> Get<Vec<T::AccountId>> for Pallet<T>
+    where
+        T: Config + pallet_session::Config,
+        // Implemented only where Session's ValidatorId is directly convertible to collator_selection's ValidatorId
+        <T as Config>::ValidatorId: From<<T as pallet_session::Config>::ValidatorId>,
+    {
+        /// Return the set of eligible collator accounts
+        fn get() -> Vec<T::AccountId>
+        where
+            <T as Config>::ValidatorId: From<<T as pallet_session::Config>::ValidatorId>,
+        {
+            use sp_runtime::traits::Convert;
+
+            let validator_account_set = pallet_session::Pallet::<T>::validators()
+                .into_iter()
+                .map(|vid: <T as pallet_session::Config>::ValidatorId| {
+                    <T as Config>::AccountIdOf::convert(vid.into())
+                })
+                .collect::<Vec<T::AccountId>>();
+
+            validator_account_set
+        }
+    }
+
+    /// Returns whether an account is part of pallet_session::Validators
+    impl<T: Config> nimbus_primitives::CanAuthor<T::AccountId> for Pallet<T> {
+        fn can_author(account: &T::AccountId, slot: &u32) -> bool {
+            let validator_key = T::ValidatorIdOf::convert(account.clone());
+            if validator_key.is_none()
+                || !T::ValidatorRegistration::is_registered(
+                    &validator_key.expect("we checked against none before. qed"),
+                )
+            {
+                return false;
+            }
+            T::CanAuthor::can_author(account, slot) // filter passed, hand execution to the next pipeline step
```

Comment:
Can this comment be more explicit, about the next and current step of the pipeline

Reply:
it actually can't, because the pallet doesn't decide the sequencing, the runtime does.

The sequencing comments are all in the calamari/dolphin lib.rs

Reply:
nit: can you mention that the order is decided by the runtime in the comment